### PR TITLE
Update dependencies & Go

### PR DESCRIPTION
**Pull request by akshayjshah**
- Update command-line build tools
- Update Go for CI
- Rename unused parameter

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go-version: [1.18.x,1.19.x]
+        go-version: [1.19.x, 1.20.x]
```
Comment (jhump) [inline .github/workflows/ci.yaml:18]: We should bump the version in `go.mod`, too.

Comment (akshayjshah) [reply .github/workflows/ci.yaml:18]: Should we? This package isn't using any language features that aren't available in 1.18, so it should compile correctly. Bumping `go.mod` _forces_ users to update, which is maybe a little extreme for routine CI maintenance?

Comment (jhump) [reply .github/workflows/ci.yaml:18]: Then shouldn't we keep `1.18.x` in the set of tested versions, to make sure we don't inadvertently add anything incompatible?

**Review by jhump — APPROVED** [review]
Approving. But if we are concerned with maintaining compatibility with Go 1.18, it would be nice to restore it to the set of tested versions.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (akshayjshah) [reply .github/workflows/ci.yaml:18]: My thinking on this has been that we only support (i.e., entertain bug reports for) the last two releases of Go, so we don't need to test any further back. If packages _happen_ to work on older versions, that's a happy coincidence for any slow-to-upgrade users.

In this particular case, though, the latest Connect doesn't support Go 1.18 anyways, so bumping `go.mod` is fine.

**Pull request merged**